docs/how-to-guides/data-basics/5-adding-a-delete-button.md

```diff
@@ -0,0 +1,454 @@
+			{ isDeleting ? (
+				<>
+					<Spinner />
+					Deleting...
+				</>
+			) : 'Delete' }
+		</Button>
+	);
+}
+```
+
+Here's what it looks like in action:
+
+![](https://raw.githubusercontent.com/WordPress/gutenberg/HEAD/docs/how-to-guides/data-basics/media/delete-button/deleting-in-progress.png)
+
+### Step 4: Handle errors
+
+We optimistically assumed that a *delete* operation would always succeed. Unfortunately, under the hood, it is a REST API request that can fail in many ways:
+
+* The website can be down.
+* The update may be invalid.
+* The page could have been deleted by someone else in the meantime.
+
+To tell the user when any of these happens, we need to extract the error information using the `getLastEntityDeleteError` selector:
+
+```js
+// Replace 9 with an actual page ID
+wp.data.select( 'core' ).getLastEntityDeleteError( 'postType', 'page', 9 )
+```
+
+Here's how we can apply it in `DeletePageButton`:
+
+```js
+import { useEffect } from '@wordpress/element';
+const DeletePageButton = ({ pageId }) => {
+	// ...
+	const { error, /* ... */ } = useSelect(
+		select => ( {
+			error: select( coreDataStore ).getLastEntityDeleteError( 'postType', 'page', pageId ),
+			// ...
+		} ),
+		[pageId]
+	);
+	useEffect( () => {
+		if ( error ) {
+			// Display the error
+		}
+	}, [error] )
+
+	// ...
+}
+```
+
+The `error` object comes from the `@wordpress/api-fetch` and contains information about the failure reason. It has the following properties:
+
+* `message` – a human-readable error message such as `Invalid post ID`.
+* `code` – a string-based error code such as `rest_post_invalid_id`. To learn about all possible error codes you'd need to refer to the [`/v2/pages` endpoint's source code](https://github.com/WordPress/wordpress-develop/blob/2648a5f984b8abf06872151898e3a61d3458a628/src/wp-includes/rest-api/endpoints/class-wp-rest-revisions-controller.php#L226-L230).
+* `data` (optional) – error details, contains the `code` property containing the HTTP response code for the failed request.
+
+There are many ways to turn that data into an error message, but in this tutorial, we will display the `error.message`.
+
+WordPress has an established pattern of displaying status information using the `Snackbar` component. Here's what it looks like **in the Widgets editor**:
+
+![](https://raw.githubusercontent.com/WordPress/gutenberg/HEAD/docs/how-to-guides/data-basics/media/delete-button/deleting-in-progress.png)
+![](./media/delete-button/snackbar-example.png)
+
+Let's use the same type of notifications in our plugin! There are two parts to this:
+
+1. Displaying notifications
+2. Dispatching notifications
+
+#### Displaying notifications
+
+Our application only knows how to display pages but does not know how to display notifications. Let's tell it!
+
+The [`Snackbar` WordPress component](https://wordpress.github.io/gutenberg/?path=/story/components-snackbar--default) represents a single notification:
```

Comment:
The jump from let's tell it! to here are two new components seems to sudden. Maybe a sentence about how we can use these components such and such would do, and then explain what eachone does.